File: labs/lab1/Python/funcs.py
import sys


def readConsole():
    text = sys.stdin.read()  # Ctrl+D для завершения ввода
    return text


def inputInFile(text, name):
    if name == 'FirstFile.txt':
        file = open(name, 'at')  # открытие на дозапись для первого файла
    else:
        file = open(name, 'wt')  # предварительное удаление контента из второго файла
    file.write(text)
    file.close()


def readFileContent(name):
    with open(name) as file:
        text = file.readlines()
    file.close()
    return text


def changeText(text):
    changed = ''
    '\n'.join(text)
    sort = sorted(text, key=len)

    for line in sort:
        line = line[:len(line) - 1]
        line += str(len(line))
        changed += line + '\n'
    changed.join('\n')
    return changed


def printFileContent(name):
    file = open(name, 'rt')
    text = file.read()
    file.close()
    for line in text:
        line.strip()
    message = ' The ' + name + ' contains: '
    print('{:~^100}'.format(message))
    print(text)

File: labs/lab1/c++/source/Funcs.cpp
#include "../headers/Funcs.h"

vector<string> ReadConsole() {
    vector<string> text;
    string line;
    int key = 35;
    while (int(line[0]) != key) {
        getline(cin, line);
        text.push_back(line);
    }
    text.pop_back();
    return text;
}

void InputInFile(vector<string> text, const string &name, fstream &myfile) {
    myfile.open(name, ios::app | ios:: ate);
    if(myfile.is_open()) {
        for (int i = 0; i < text.size(); i++) {
            myfile << text[i] << endl;
        }
    }
    myfile.close();
}

void ChangeText(vector<string> &text) {
    for (int i = 0; i <  text.size(); i++) {
        for (int j = i + 1; j < text.size(); j++) {
            if (text[i].size() > text[j].size()) {
                string temp = text[i];
                text[i] = text[j];
                text[j] = temp;
            }
        }
    }
    for (int i = 0; i < text.size(); i++) {
        text[i] += " " + to_string(text[i].size());
    }
}

void PrintFileContent(const string& message, const string &name, fstream &myfile) {
    myfile.open(name, ios::in);
    string line;
    cout << message << endl;
    while (getline(myfile, line)) {
        cout << line << endl;
    }
    myfile.close();
}


